### ISSUE TITLE
[doc] Broken Links

### ISSUE THREAD

**Issue by HU90m** [lowRISC/opentitan#17658]
### Description

In the documentation, there are a few broken links to external sites that no longer exist and without straight forward replacements,.

* <https://www.nxp.com/docs/en/user-guide/UM10204.pdf> in `./hw/ip/i2c/README.md` and `./hw/ip/i2c/doc/theory_of_operation.md`.
* <https://www.sogis.eu/documents/cc/crypto/SOGIS-Agreed-Cryptographic-Mechanisms-1.1.pdf> in `./doc/security/specs/identities_and_root_keys/README.md`.
* <https://opentitan.org/book/hw/top_earlgrey/ip/pinmux/doc/autogen/pinout_asic/> in `./site/landing/layouts/shortcodes/earlgrey-diagram.html`.
* In `./doc/security/implementation_guidelines/hardware/README.md`
 * <https://www.riscure.com/uploads/2020/05/Riscure_Whitepaper_Fault_Mitigation_Patterns_final.pdf>
 * <https://ts.data61.csiro.au/projects/TS/timeprotection/> which could maybe replaced with <https://trustworthy.systems/publications/full_text/Ge_YCH_19.pdf>
* <https://riscv.org/specifications/debug-specification> in `./doc/contributing/dv/methodology/README.md`.

**Comment by hcallahan-lowrisc** [comment]
> * https://www.nxp.com/docs/en/user-guide/UM10204.pdf in `./hw/ip/i2c/README.md` and `./hw/ip/i2c/doc/theory_of_operation.md`.

For this one at-least, Marno and I were discussing using a wayback-machine link. (The document is still available on nxp.com, but now behind a registration. e.g. https://www.nxp.com/webapp/Download?colCode=UM10204)

**Comment by marnovandermaas** [comment]
> > * https://www.nxp.com/docs/en/user-guide/UM10204.pdf in `./hw/ip/i2c/README.md` and `./hw/ip/i2c/doc/theory_of_operation.md`.
> 
> For this one at-least, Marno and I were discussing using a wayback-machine link. (The document is still available on nxp.com, but now behind a registration. e.g. https://www.nxp.com/webapp/Download?colCode=UM10204)

Yes, I have a fix for this in this PR: https://github.com/lowRISC/opentitan/pull/17435
Need to rebase that since the IP documentation has been split.

**Comment by marnovandermaas** [comment]
Some replacement suggestions:
Sogis, there is a [1.2 spec](https://www.sogis.eu/documents/cc/crypto/SOGIS-Agreed-Cryptographic-Mechanisms-1.2.pdf) out. We need to check whether 1.2 still applies to our documentation. The internet archive does not have a captured version of the 1.1 spec unfortunately.
Riscure whitepaper: https://web.archive.org/web/20220616035354/https://www.riscure.com/uploads/2020/05/Riscure_Whitepaper_Fault_Mitigation_Patterns_final.pdf
RISC-V debug spec: https://raw.githubusercontent.com/riscv/riscv-debug-spec/release/riscv-debug-release.pdf
